### ISSUE TITLE
Consider updating to StackExchange.Redis 2.1

### ISSUE THREAD

**Issue by erizzo** [MichaCo/CacheManager#309]
StackExchange.Redis is now at version 2.1.x, and there are [useful fixes](https://stackexchange.github.io/StackExchange.Redis/ReleaseNotes) in there. Most notably to me is fixing some intermittent [connectivity problems with Azure Redis](https://github.com/StackExchange/StackExchange.Redis/pull/1374). That is an issue that my team has been seeing over the past year or so and caused me some sleepless nights (I work on a very large platform for a very large enterprise client, heavily invested in Azure).
Will you consider releasing an updated version of CacheManager.StackExchange.Redis? If I was willing to submit a PR would that change your answer?

**Comment by MichaCo** [comment]
Is there any breaking change in the 2.1 client which prevents you from using that version together with CacheManager?

**Comment by erizzo** [comment]
Do you mean referencing the StackExchange.Redis Nuget package directly, essentially overriding the reference in CacheManager?
We thought about that but aren't sure it will actually work. Assuming StackExchange follows semantic versioning, it's just a minor update so should be backwards-compatible.

**Comment by MichaCo** [comment]
Yeah exactly. That's usually how you do that with Nugets, same as with .NET frameworks, I usually target slightly lower versions in libraries so that I don't force consumers to upgrade.

And, over time, dependencies might get new versions and I don't want/have to release a new version of a library just because one dependency got a new minor version.

Just give it a try.

**Comment by keithl8041** [comment]
Can confirm that we're running CacheManager 1.2 with SE.Redis 2.1 in production and haven't seen any issues yet.

**Comment by erizzo** [comment]
@keithl8041 Sounds similar to our issues, yes. Although we see it across a bunch of different application services (web APIs) and I'm not sure of the deployment structure, whether the affected APIs at any time are in the same app service instance or not. I'll ask our devops folks about that the next time it's reported/observed.

**Comment by tonyqus** [comment]
We are encountering an existing a bug in StackExchange.Redis on production
https://github.com/StackExchange/StackExchange.Redis/issues/1120

We are using the version 2.0.0-beta-1629 with StackExchange.Redis 2.0.513

If you can create a new release based on StackExchange.Redis 2.1 version, it would work for us.

**Comment by borgmatthew** [comment]
Updated the version of StackExchange.Redis in #337
